drivers/net/wireless/mediatek/mt76/mt76_connac_mac.c

```diff
@@ -13,6 +13,9 @@ int mt76_connac_pm_wake(struct mt76_phy *phy, struct mt76_connac_pm *pm)
 	if (mt76_is_usb(dev))
 		return 0;
 
+	if (test_bit(MT76_STATE_RESET, &phy->state))
+		return 0;
+
```

Comment:
@objelf I guess this approach is a bit racy because we can run drv_own or fw_own while running the reset right?

Reply:
drv_own and fw_own don't work during wifisys reset

Reply:
in the code flowpath we can have both mt7921_reset and fw_own/drv_own running at the same time, right (e.g. on different CPUs). e.g what happen if mt76_connac_pm_wake has already scheduled the wake work?

Reply:
@objelf sorry this comment is related to the change "https://github.com/LorenzoBianconi/wireless-drivers-next/pull/33/commits/0712b8987b3f89947954f9a8819fe890b785c453". Please ignore it